frontera/contrib/backends/cassandra/components.py

```diff
@@ -0,0 +1,314 @@
+        for fprint, score, request, schedule in batch:
+            cql_i = (score, self.crawl_id, fprint)
+            cql_items.append(cql_i)
+        execute_concurrent_with_args(self.session, query, cql_items, concurrency=400)
+        self.counter_cls.cass_count({"scored_urls": len(cql_items)})
+
+
+class States(MemoryStates):
+
+    def __init__(self, session, model_cls, cache_size_limit, crawl_id):
+        super(States, self).__init__(cache_size_limit)
+        self.session = session
+        self.model = model_cls
+        self.table = 'StateModel'
+        self.logger = logging.getLogger("frontera.contrib.backends.cassandra.components.States")
+        self.crawl_id = crawl_id
+
+    def frontier_stop(self):
+        pass
+
+    def fetch(self, fingerprints):
+        to_fetch = [f for f in fingerprints if f not in self._cache]
+        self.logger.debug("cache size %s", len(self._cache))
+        self.logger.debug("to fetch %d from %d", (len(to_fetch), len(fingerprints)))
+
+        for chunk in chunks(to_fetch, 128):
+            for state in self.model.objects.filter(crawl=self.crawl_id, fingerprint__in=chunk):
+                self._cache[state.fingerprint] = state.state
+
+    def flush(self, force_clear=False):
+        query = self.session.prepare("INSERT INTO states (crawl, fingerprint, state) VALUES (?, ?, ?)")
+        cql_items = []
+        for fingerprint, state_val in self._cache.iteritems():
+            cql_i = (self.crawl_id, fingerprint, state_val)
+            cql_items.append(cql_i)
+        execute_concurrent_with_args(self.session, query, cql_items, concurrency=20000)
+        super(States, self).flush(force_clear)
+
+
+class Queue(BaseQueue):
+    def __init__(self, session, queue_cls, partitions, crawl_id, generate_stats, ordering='default'):
+        self.session = session
+        self.queue_model = queue_cls
+        self.logger = logging.getLogger("frontera.contrib.backends.cassandra.components.Queue")
+        self.partitions = [i for i in range(0, partitions)]
+        self.partitioner = Crc32NamePartitioner(self.partitions)
+        self.ordering = ordering
+        self.crawl_id = crawl_id
+        self.counter_cls = CassandraCount(crawl_id, self.session, generate_stats)
+
+    def frontier_stop(self):
+        pass
+
+    def _order_by(self):
+        if self.ordering == 'created':
+            return "created_at"
+        return "created_at"
+
+    def get_next_requests(self, max_n_requests, partition_id, **kwargs):
+        """
+        Dequeues new batch of requests for crawling.
+
+        :param max_n_requests: maximum number of requests to return
+        :param partition_id: partition id
+        :return: list of :class:`Request <frontera.core.models.Request>` objects.
+        """
+        results = []
+        try:
+            dequeued_urls = 0
+            cql_ditems = []
+            d_query = self.session.prepare("DELETE FROM queue WHERE crawl = ? AND fingerprint = ? AND partition_id = ? "
```

Comment:
To Delete an column in cassandra i must provide the keys in an specific order. I cannot make a deletion WHERE fingerprint='' - cassandra needs the keys in the roght order bevor this key. I also cannot move fingerprint to the beginning of the table because the same i have to do by an order by clause...
Perhaps i can make an query with fingerprint IN ... to make only one query... But it seems that that is not much faster...